rust/examples/src/bin/simple_market_maker.rs

```diff
@@ -0,0 +1,224 @@
+use anyhow::anyhow;
+use borsh::BorshDeserialize;
+use phoenix::{
+    program::create_new_order_instruction,
+    quantities::{BaseLots, Ticks, WrapperU64},
+    state::{OrderPacket, Side},
+};
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_program::{clock::Clock, sysvar};
+use solana_sdk::signature::Signature;
+use std::str::FromStr;
+
+use clap::Parser;
+use phoenix_sdk::sdk_client::{MarketMetadata, SDKClient};
+use solana_cli_config::{Config, CONFIG_FILE};
+#[allow(unused_imports)]
+use solana_sdk::{
+    instruction::Instruction,
+    program_pack::Pack,
+    pubkey::Pubkey,
+    signature::{read_keypair_file, Keypair},
+    signer::Signer,
+};
+pub mod utils;
+use crate::utils::marinade::MarinadeState;
+
+#[derive(Parser)]
+struct Args {
+    #[clap(short, long)]
+    url: Option<String>,
+    #[clap(short, long)]
+    market: Pubkey,
+    #[clap(short, long, default_value = "SOL-USD")]
+    ticker: String,
+    /// Optionally include your keypair path. Defaults to your Solana CLI config file.
+    #[clap(short, long)]
+    keypair_path: Option<String>,
+    #[clap(long, default_value = "2000")]
+    quote_refresh_frequency_in_ms: u64,
+    #[clap(long, default_value = "5")]
+    quote_edge_bps: u64,
+    #[clap(long, default_value = "250.0")]
+    quote_size_in_quote_units: f64,
+    #[clap(long, default_value = "10")]
+    order_lifetime_in_seconds: i64,
+}
+
+pub fn get_payer_keypair_from_path(path: &str) -> anyhow::Result<Keypair> {
+    read_keypair_file(&*shellexpand::tilde(path)).map_err(|e| anyhow!(e.to_string()))
+}
+
+pub fn get_network(network_str: &str) -> &str {
+    match network_str {
+        "devnet" | "dev" | "d" => "https://api.devnet.solana.com",
+        "mainnet" | "main" | "m" | "mainnet-beta" => "https://api.mainnet-beta.solana.com",
+        "localnet" | "localhost" | "l" | "local" => "http://localhost:8899",
+        _ => network_str,
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    // Connect to the solana network and get the market address
+    let args = Args::parse();
+    let config = match CONFIG_FILE.as_ref() {
+        Some(config_file) => Config::load(config_file).unwrap_or_else(|_| {
+            println!("Failed to load config file: {}", config_file);
+            Config::default()
+        }),
+        None => Config::default(),
+    };
+    let trader = get_payer_keypair_from_path(&args.keypair_path.unwrap_or(config.keypair_path))?;
```

Comment:
kinda wack that these override each other

Reply:
would you prefer no override and instead an error?